.github/workflows/pr_test_when_merged.yml

```diff
@@ -26,7 +26,7 @@ jobs:
         uses: ./
         id: test-pipeline
   publish:
-    needs: [ run_test_on_testnet, run_test_on_mainnet ]
```

Comment:
why run_test_on_testnet is removed?

Reply:
The `needs` field means that any jobs written in this field must complete successfully before this job will run.
If any of the listed jobs fails or is skipped, the publish phase will also be skipped.

However, in our pipeline, either `run_test_on_testnet` or `run_test_on_mainnet` is skipped unconditionally.
1) merging PR in develop branch: `run_test_on_mainnet` is skipped and only `run_test_on_testnet` runs.
1) merging PR in master branch: `run_test_on_testnet` is skipped and only `run_test_on_mainnet` runs.

i removed the run_test_on_testnet because the `publish` step only requires the success of `run_test_on_mainnet` job.

Reply:
(discussed offline) thanks for the explanation!